homeassistant/components/http/ban.py

```diff
@@ -105,18 +105,18 @@ async def process_wrong_login(request):
     except herror:
         pass
 
-    msg = f"Login attempt or request with invalid authentication from {remote_host} ({remote_addr})"
+    base_msg = f"Login attempt or request with invalid authentication from {remote_host} ({remote_addr})."
 
-    user_agent_raw = request.headers.get("user-agent")
-    if user_agent_raw:
-        # Sanitize user agent to limited charset
-        user_agent = re.sub(r"[^a-zA-Z0-9:;,./() ]", "_", user_agent_raw)
-        msg = f"{msg} ({user_agent})"
+    # The user-agent is unsanitized input so we only include it in the log
+    user_agent = request.headers.get("user-agent")
+    log_msg = f"{base_msg} ({user_agent})"
```

Comment:
The test whether the user-agent is set is removed on purpose. For the log its better that the message format is stable, and the None/Empty that is printed when not set is potentially useful for the reader.